application/app/utils/providers/initialViewProvider.tsx

```diff
@@ -0,0 +1,25 @@
+'use client';
+
+import { ReactNode, createContext, useCallback, useState } from 'react';
+
+export interface InitialViewContextType {
+	isInitialView: boolean;
+	closeInitialView: () => void;
+}
+
+export const InitialViewContext = createContext<InitialViewContextType | null>(null);
+
+//TODO: if useURLParams has values then don't show initial view
+export const InitialViewProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+	const [isInitialView, setIsInitialView] = useState(true);
+
+	const closeInitialView = useCallback(() => {
+		setIsInitialView(false);
+	}, []);
+
+	return (
+		<InitialViewContext.Provider value={{ isInitialView, closeInitialView }}>
+			{children}
+		</InitialViewContext.Provider>
+	);
+};
```

Comment:
Tu peux mettre en `undefined` le context par defaut dans createContext puis creer un hook pour renvoyer une erreur si le hook est appele en dehors du provider (donc undefined)

```suggestion
const InitialViewContext = createContext<InitialViewContextType | undefined>(undefined);

//TODO: if useURLParams has values then don't show initial view
export const InitialViewProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
	const [isInitialView, setIsInitialView] = useState(true);

	const closeInitialView = useCallback(() => {
		setIsInitialView(false);
	}, []);

	return (
		<InitialViewContext.Provider value={{ isInitialView, closeInitialView }}>
			{children}
		</InitialViewContext.Provider>
	);
};

export function useInitialView() {
	const context = useContext(InitialViewContext);

	if (context === undefined) {
		throw new Error('useInitialView must be used with a InitialViewProvider');
	}

	return context;
}
```